dependencies.yaml

```diff
@@ -115,7 +115,7 @@ dependencies:
               - cuda-version=11.8
               - cudatoolkit
           - matrix:
-              cuda: "12.0"
+              cuda: "12.*"
             packages:
               - cuda-version=12.0
               - cuda-cudart-dev
```

Comment:
This isn't right. We shouldn't be installing `cuda-version=12.0` for all `cuda 12.*` inputs.